crossdock/src/helpers.js

```diff
@@ -20,10 +20,196 @@
 // THE SOFTWARE.
 
 import * as constants from './constants.js';
+import fs from 'fs';
+import path from 'path';
+import dns from 'dns';
+import opentracing from 'opentracing';
+import request from 'request';
+import RSVP from 'rsvp';
+import Span from '../../src/span.js';
+import SpanContext from '../../src/span_context.js';
+import TChannel from 'tchannel/channel';
+import TChannelThrift from 'tchannel/as/thrift';
+import TChannelBridge from '../../src/tchannel_bridge';
 import Utils from '../../src/util.js';
 
 export default class Helpers {
-    static observeSpan(context: any): ObservedSpan {
+    _tracer: Tracer;
+    _bridge: TChannelBridge;
+    _thriftChannel: Channel;
+
+    constructor(tracer: Tracer) {
+        this._tracer = tracer;
+
+        var channel = TChannel().makeSubChannel({
+            serviceName: 'node',
+            peers: [Utils.myIp() + ':8082']
+        });
+
+        let crossdockSpec = fs.readFileSync(path.join(__dirname, '..', '..', 'src', 'jaeger-idl', 'thrift', 'crossdock', 'tracetest.thrift'), 'utf8');
+        this._thriftChannel = TChannelThrift({
+            channel: channel,
+            source: crossdockSpec
+        });
+
+        let bridge = new TChannelBridge(this._tracer);
+        this._tracedChannel= bridge.tracedChannel(this._thriftChannel);
+    }
+
+    handleRequest(isStartRequest: boolean, traceRequest: any, spanOrContext: any, operationName: string): void {
+        let serverSpan;
+        if (spanOrContext instanceof Span) {
+            serverSpan = spanOrContext;
+        }
+
+        if (spanOrContext instanceof SpanContext) {
+            serverSpan = this._tracer.startSpan(operationName, { childOf: spanOrContext });
+        }
+
+        if (isStartRequest) {
+            serverSpan.setBaggageItem(constants.BAGGAGE_KEY, traceRequest.baggage);
+            if (traceRequest.sampled) {
+                serverSpan.setTag(opentracing.Tags.SAMPLING_PRIORITY, 1);
+            }
+        }
+
+        // do async call to prepareResponse
+        return new RSVP.Promise((resolve, reject) => {
+            this.prepareResponse(traceRequest.downstream, {span: serverSpan}).then((response) => {
```

Comment:
not following this - prepareResponse already returns a promise, you just need to pass it up, not wrap it.